File: packages/plplot/examples/python/x12.py
#!/usr/bin/env python
#
#	x12c.c
#
#	Bar chart demo.

import pl
import sys
import os

module_dir = "@MODULE_DIR@"

if module_dir[0] == '@':
	module_dir = os.getcwd ()

sys.path.insert (0, module_dir)

# main
#
# Does a simple bar chart, using color fill.  If color fill is
# unavailable, pattern fill is used instead (automatic).

def main(w):

##    # Parse and process command line arguments
##
##    pl.ParseOpts(sys.argv, pl.PARSE_FULL)
##
##    # Initialize plplot
##
##    pl.init()

    w.pladv(0)
    w.plvsta()
    w.plwind(1980.0, 1990.0, 0.0, 35.0)
    w.plbox("bc", 1.0, 0, "bcnv", 10.0, 0)
    w.plcol(2)
    w.pllab("Year", "Widget Sales (millions)", "#frPLplot Example 12")

    y0 = [5, 15, 12, 24, 28, 30, 20, 8, 12, 3]

    for i in range(10):
	w.plcol(i + 1)
	w.plpsty(0)
	fbox( w, (1980. + i), y0[i] )
	string = `y0[i]`
	w.plptex((1980. + i + .5), (y0[i] + 1.), 1.0, 0.0, .5, string)
	string = `1980 + i`
	w.plmtex("b", 1.0, ((i + 1) * .1 - .05), 0.5, string)

    # Don't forget to call PLEND to finish off!

    w.pleop()

def fbox( w, x0, y0 ):

    x = [x0, x0, x0 + 1., x0 + 1.]
    y = [0., y0, y0, 0.]
    w.plfill(x, y)
    w.plcol(1)
    w.pllsty(1)
    w.plline(x, y)

##main()

File: packages/plplot/examples/perl/x12.pl
#! /usr/bin/env perl
#
# Demo x12 for the PLplot PDL binding
#
# Bar chart demo
#
# Copyright (C) 2004  Rafael Laboissiere
#
# This file is part of PLplot.
#
# PLplot is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Library Public License as published
# by the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# PLplot is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU Library General Public License for more details.
#
# You should have received a copy of the GNU Library General Public License
# along with PLplot; if not, write to the Free Software
# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA

# SYNC: x12c.c 1.14

use PDL;
use PDL::Graphics::PLplot;

# Parse and process command line arguments

plParseOpts (\@ARGV, PL_PARSE_SKIP | PL_PARSE_NOPROGRAM);

# Initialize plplot

plinit ();

pladv (0);

plvsta ();
plwind (1980.0, 1990.0, 0.0, 35.0);
plbox (1.0, 0, 10.0, 0, "bc", "bcnv");
plcol0 (2);
pllab ("Year", "Widget Sales (millions)", "#frPLplot Example 12");

my $y0 = pdl [5, 15, 12, 24, 28, 30, 20, 8, 12, 3];

for (my $i = 0; $i < 10; $i++) {
  plcol0 ($i + 1);
  plpsty (0);
  myplfbox ((1980. + $i), $y0->index ($i));
  my $string = sprintf ("%.0f", $y0->index ($i));
  plptex ((1980. + $i + .5), ($y0->index ($i) + 1.), 1.0, 0.0, .5, $string);
  $string = sprintf ("%d", 1980 + i);
  plmtex (1.0, (($i + 1) * .1 - .05), 0.5, "b", $string);
}

# Don't forget to call plend() to finish off!

plend ();

# The subroutine below is called plfbox in the x12c.c examples, but
# the PDL binding define already a plfbos that conflicts with this
# one.  Hence, it was recalled myplfbox here.

sub myplfbox {
  my $x0 = shift;
  my $y0 = shift;

  my $x = $x0 + pdl [0, 0, 1, 1];
  my $y = pdl [0, $y0->sclr, $y0->sclr, 0];

  plfill ($x, $y);
  plcol0 (1);
  pllsty (1);
  plline ($x, $y);
}
